runtime/dynlink.c

```diff
@@ -229,6 +230,55 @@ void caml_free_shared_libs(void)
     caml_dlclose(shared_libs.contents[--shared_libs.size]);
 }
 
+static value list_of_ext_table(struct ext_table* tbl)
+{
+  CAMLparam0();
+  CAMLlocal3(list, tmp, str);
+  int i;
+  for (i = tbl->size - 1; i >= 0; i--) {
+    str = caml_copy_string(tbl->contents[i]);
+    tmp = caml_alloc_small(2, 0);
+    Field(tmp, 0) = str;
+    Field(tmp, 1) = list;
+    list = tmp;
+  }
+  CAMLreturn (list);
+}
+
+CAMLprim value caml_dynlink_get_bytecode_sections(value unit)
```

Comment:
If this primitive were to treated as "once only", we could free all the memory at the end of the call to it?

Reply:
Actually, having mused on it further, there's a fairly easy scheme related to the `CRCS` section which would mean that the `SYMB` and `CRCS` would only be read at start-up if the program were then going to use `Dll`/`Meta`. That would also slightly simplify the bootstrap process. Given your benchmarking, I think that could _all_ therefore be left to a subsequent PR.